### PR TITLE
Fix jax.random.poisson(lam=inf) returning 0 instead of dtype max

### LINKED ISSUE #38335: jax.random.poisson(lam=inf) returns 0 (the lam=0 answer); huge finite lam saturates to INT64_MAX
### Description

`lam=inf` is inside the documented domain (`lam >= 0`), and the Poisson distribution diverges as lam → inf. JAX returns **0** — the highest-probability sample of the opposite extreme lam=0 — deterministically, for every seed:

```python
import jax, numpy as np
jax.config.update('jax_enable_x64', True)
key = jax.random.PRNGKey(0)
print(jax.random.poisson(key, np.inf, (3,)))   # [0 0 0]
print(jax.random.poisson(key, 1e30, (3,)))     # [9223372036854775807 ...]
```

| lam | result (int64, x64) |
|---|---|
| 1e10 | ~1e10 (correct scale) |
| 1e18 | ~1e18 |
| 1e30 … float32.max | 9223372036854775807 (INT64_MAX saturation) |
| **inf** | **0** |
| nan | −1 |

The discontinuity is exactly at inf — every huge finite lam saturates sensibly, then inf collapses to the one value that is indistinguishable from a legitimate small-lam sample.

### Root cause

In `jax/_src/random/core.py::_poisson` (same logic in 0.10.1's `random.py`), `lam=inf` routes to `_poisson_rejection` (`use_knuth = isnan(lam) | (lam < 10)` is False), where

```python
k = lax.floor((2 * a / u_shifted + b) * u + lam + 0.43)   # = inf for lam=inf
...
return ... k.astype(dtype)
```

converts float inf to int64 — undefined behavior in XLA, which lands on 0 on CPU. Finite-huge lam converts a finite float and saturates to INT64_MAX, hence the clean behavior up to float32.max. `_poisson` special-cases only `lam == 0`; nothing handles non-finite lam (the nan → −1 is itself accidental: nan comparisons fail in the Knuth loop and `k − 1` falls out with k=0).

### Suggested fix

Extend the final select, e.g. `lax.select(lam == np.inf, full_like(result, iinfo(dtype).max), result)` — or route non-finite lam to the −1 sentinel like nan. Either is defensible; a silent 0 is not.

NumPy raises `ValueError` for `np.random.poisson(np.inf)`; since JAX can't raise under jit, a sentinel/saturation convention is the appropriate analogue.

### System info (python version, jaxlib version, accelerator, etc.)

```
jax: 0.10.1 (root cause verified unchanged on current main, jax/_src/random/core.py L1816-1833)
jaxlib: 0.10.1, numpy 2.4.6, python 3.12.13, CPU (arm64, macOS)
```

### PR DESCRIPTION
**Pull request by factnn**
## Summary

Fixes #38335: `jax.random.poisson(lam=inf)` returns `0` instead of `INT64_MAX`.

## Root Cause

`lam=inf` → `_poisson_rejection` path → `k = floor(... + inf + 0.43) = inf` → `float(inf)` cast to `int64` is undefined behavior in XLA, which lands on `0` on CPU.

## Fix

Saturate to `dtype max` (e.g., `INT64_MAX`), consistent with the behavior of huge finite lam values which also saturate to `INT64_MAX`.

## Changes

- `jax/_src/random/core.py`: 5 lines — add `lax.isinf(lam)` check after `lam==0` special case
- `tests/random_lax_test.py`: 8 lines — regression test

Fixes #38335

* random
  * Fixed `poisson(lam=inf)` returning 0 instead of dtype max ([#38335](https://github.com/jax-ml/jax/issues/38335))

### CODE REVIEW

**Comment by factnn** [comment]
@jakevdp FYI I've closed all the scipy-related PRs (#38267, #38288) to not waste your time and energy on those edge cases. The remaining 5 open PRs (#38001, #38012, #38029, #38030, #38459) are not scipy compatibility fixes — they're real bugs from 5 different users, targeting actual issues in the JAX repo. Would appreciate a look when you have time.

**Comment by jakevdp** [comment]
This falls under #38651; please note that it may take a while for this to receive attention from the team.
